noir-projects/aztec-nr/aztec/src/history/test.nr

```diff
@@ -0,0 +1,34 @@
+use crate::{note::lifecycle::{create_note, destroy_note}, oracle::execution::get_contract_address};
+use crate::note::retrieved_note::RetrievedNote;
+use crate::test::{helpers::test_environment::TestEnvironment, mocks::mock_note::MockNote};
+
+// We create a note at block 2 and nullify it in the next block. This helps us test our historical apis.
+pub(crate) unconstrained fn create_note_and_optionally_nullify_it(
+    nullify_note: bool,
```

Comment:
Instead of having this flag I think it would be cuter if you had 2 functions:

1. create_note
2. create_note_and_nullify_it

Then a reader would not have to bother with parsing what is the flag when looking at tests not using nullified notes.

WDYT?